ios/NotificationService/NotificationService.swift

```diff
@@ -14,38 +15,69 @@ class NotificationService: UNNotificationServiceExtension {
       let type = bestAttemptContent.userInfo["type"]
       let postId = bestAttemptContent.userInfo["post_id"]
       let idLoaded = bestAttemptContent.userInfo["id_loaded"] ?? false
+
       UploadSession.shared.notificationReceipt(
         notificationId: ackId,
-        receivedAt: Date().millisencondsSince1970,
+        receivedAt: Date().millisecondsSince1970,
         type: type,
         postId: postId,
         idLoaded: idLoaded as! Bool
       ) { data, error in
         if (idLoaded as! Bool) {
           guard let data = data, error == nil else {
-            return
-          }
-
-          let json = try? JSONSerialization.jsonObject(with: data) as! [String: Any]
-          if let json = json {
-            if let message = json["message"] as? String {
-              bestAttemptContent.body = message
-            }
-            if let channelName = json["channel_name"] as? String {
-              bestAttemptContent.title = channelName
-            }
-
-            let userInfoKeys = ["channel_name", "team_id", "sender_id", "root_id", "override_username", "override_icon_url", "from_webhook"]
-            for key in userInfoKeys {
-              if let value = json[key] as? String {
-                bestAttemptContent.userInfo[key] = value
+            self.sendFailed = true;
+            let fibonacciBackoffsInSeconds = [1.0, 2.0, 3.0, 5.0, 8.0]
+            for backoffInSeconds in fibonacciBackoffsInSeconds {
+              let timer = Timer.scheduledTimer(withTimeInterval: backoffInSeconds, repeats: false) { timer in
+                UploadSession.shared.notificationReceipt(
```

Comment:
Is there a way, maybe with recursion, so you don’t need to write this block twice? Another option would be having trigger on 0 seconds

Reply:
Yeah, I wasn't happy with this either, and the alternatives (e.g passing the closure as a callback to the original call) were taking a bit of time until I have a better grasp of the syntax.

Perhaps I can handle this refactoring as a quick, "fast follow" task?

Reply:
fine by me